### PR TITLE
refactor(web): add-utility-to-check-empty-strings-with-whitespaces

### PR DESCRIPTION
**Pull request by Harman-singh-waraich**
## PR-Codex overview
This PR introduces a new utility function `isEmpty` to check if a string is empty or contains only whitespace. It replaces instances of direct string checks with `isEmpty` for improved readability and consistency across various components.

### Detailed summary
- Added `isEmpty` function in `web/src/utils/index.ts`.
- Updated `display` logic in `StyledButton` to use `isEmpty` in `PreviousButton.tsx`.
- Modified condition in `FormContact.tsx` to use `isEmpty` for `contactInput`.
- Changed `isDisabled` logic in `SubmitEvidenceModal.tsx` to use `isEmpty` for `message`.
- Used `isEmpty` in `Search.tsx` for `search` checks.
- Updated `policyURI` check in `NewDisputeContext.tsx` to use `isEmpty`.
- Refactored alias validation logic in `NextButton.tsx` to use `isEmpty`.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

## Summary by CodeRabbit

- **New Features**
	- Introduced a new utility function, `isEmpty`, to enhance string validation across various components.
	- Updated logic in multiple components to use `isEmpty` for improved checks on empty strings and better control flow.

- **Bug Fixes**
	- Enhanced validation logic for fields such as `policyURI` and input fields in forms, ensuring more robust checks.

- **Documentation**
	- Updated interface to include an optional `aliasesArray` in dispute data structure.

These changes improve the overall functionality and user experience when interacting with forms and navigation within the application.

### CODE REVIEW

File: web/src/layout/Header/navbar/Menu/Settings/Notifications/FormContactDetails/FormContact.tsx [hunk 1]
```diff
@@ -47,7 +48,7 @@ const FormContact: React.FC<IForm> = ({
   };
 
   const fieldVariant = useMemo(() => {
-    if (contactInput === "" || !isEditing) {
+    if (isEmpty(contactInput) || !isEditing) {
       return undefined;
     }
     return contactIsValid ? "success" : "error";
```
[the remark this answers is not in the document]
Comment (Harman-singh-waraich) [reply web/src/layout/Header/navbar/Menu/Settings/Notifications/FormContactDetails/FormContact.tsx:54]: It's an OR condition , how does the order change it?

File: web/src/pages/Resolver/NavigationButtons/NextButton.tsx [hunk 2]
```diff
@@ -5,6 +5,7 @@ import { useLocation, useNavigate } from "react-router-dom";
 import { Button } from "@kleros/ui-components-library";
 
 import { useNewDisputeContext } from "context/NewDisputeContext";
+import { isEmpty } from "src/utils";
```
[the remark this answers is not in the document]
Comment (Harman-singh-waraich) [reply web/src/pages/Resolver/NavigationButtons/NextButton.tsx:8]: It's fine

File: web/src/context/NewDisputeContext.tsx [hunk 3]
```diff
@@ -133,7 +133,7 @@ const constructDisputeTemplate = (disputeData: IDisputeData) => {
   for (const answer of baseTemplate.answers) {
     answer.id = "0x" + BigInt(answer.id).toString(16);
   }
-  if (!isUndefined(baseTemplate.policyURI) && baseTemplate.policyURI === "") delete baseTemplate.policyURI;
+  if (!isUndefined(baseTemplate.policyURI) && isEmpty(baseTemplate.policyURI)) delete baseTemplate.policyURI;
```
[the remark this answers is not in the document]
Comment (Harman-singh-waraich) [reply web/src/context/NewDisputeContext.tsx:136]: leaving them undefined might cause problem with schema check later on. So deleting them is preffered.

File: web/src/pages/Cases/CaseDetails/Evidence/SubmitEvidenceModal.tsx [hunk 4]
```diff
@@ -14,7 +14,7 @@
 
 import EnsureAuth from "components/EnsureAuth";
 import { EnsureChain } from "components/EnsureChain";
-import { isUndefined } from "src/utils";
+import { isEmpty, isUndefined } from "src/utils";
```
[the remark this answers is not in the document]
Comment (Harman-singh-waraich) [reply web/src/pages/Cases/CaseDetails/Evidence/SubmitEvidenceModal.tsx:17]: Tis fine

File: web/src/utils/index.ts [hunk 5]
```diff
@@ -1,2 +1,7 @@
 export const isUndefined = (maybeObject: any): maybeObject is undefined | null =>
   typeof maybeObject === "undefined" || maybeObject === null;
+
+/**
+ * Checks if a string is empty or contains only whitespace.
+ */
+export const isEmpty = (str: string): boolean => str.trim() === "";
```
[the remark this answers is not in the document]
Comment (Harman-singh-waraich) [reply web/src/utils/index.ts:7]: I meant to keep it simple. This function can be paired with isUndefined if required.

File: web/src/layout/Header/navbar/Menu/Settings/Notifications/FormContactDetails/FormContact.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (Harman-singh-waraich) [reply web/src/layout/Header/navbar/Menu/Settings/Notifications/FormContactDetails/FormContact.tsx:54]: I see what you mean 👍

File: web/src/layout/Header/navbar/Menu/Settings/Notifications/FormContactDetails/FormContact.tsx [hunk 6]
```diff
@@ -2,6 +2,7 @@
 import styled from "styled-components";
 
 import { Field } from "@kleros/ui-components-library";
+import { isEmpty } from "src/utils";
```
[the remark this answers is not in the document]
Comment (Harman-singh-waraich) [reply web/src/layout/Header/navbar/Menu/Settings/Notifications/FormContactDetails/FormContact.tsx:5]: In those place it is paired with isUndefined to make sure it's not undefined. So the argument passed in will be defined, as it is checked by the previous function.

**Review by alcercu — APPROVED** [review]
lgtm

**Pull request merged**